pkg/subctl/cmd/verify_connectivity.go

```diff
@@ -0,0 +1,89 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner/test/e2e"
+	_ "github.com/submariner-io/submariner/test/e2e/dataplane"
+	"github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var (
+	verboseConnectivityVerification bool
+	operationTimeout                uint
+	connectionTimeout               uint
+	connectionAttempts              uint
+	reportDirectory                 string
+)
+
+func init() {
+	verifyConnectivityCmd.Flags().BoolVar(&verboseConnectivityVerification, "verbose", false, "Produce verbose logs during connectivity verification")
+	verifyConnectivityCmd.Flags().UintVar(&operationTimeout, "operation-timeout", 240, "Operation timeout for K8s API calls")
+	verifyConnectivityCmd.Flags().UintVar(&connectionTimeout, "connection-timeout", 60, "Connection timeout for every check")
+	verifyConnectivityCmd.Flags().UintVar(&connectionAttempts, "connection-attempts", 2, "Connection attempts for every check")
+	verifyConnectivityCmd.Flags().StringVar(&reportDirectory, "report-dir", ".", "XML report directory")
+
+	rootCmd.AddCommand(verifyConnectivityCmd)
+}
+
+var verifyConnectivityCmd = &cobra.Command{
+	Use:   "verify-connectivity",
+	Short: "Verify connectivity between two clusters",
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		err := checkValidateArguments(args)
+		exitOnError("Argument missing", err)
+		configureTestingFramework(args)
+		e2e.RunE2ETests(&testing.T{})
+	},
+}
+
+func configureTestingFramework(args []string) {
+	framework.TestContext.KubeConfigs = args
+	framework.TestContext.OperationTimeout = operationTimeout
+	framework.TestContext.ConnectionTimeout = connectionTimeout
+	framework.TestContext.ConnectionAttempts = connectionAttempts
+	framework.TestContext.ReportDir = reportDirectory
+	framework.TestContext.ReportPrefix = "subctl"
+
+	// For some tests this is only printing, but in some of them they need those to be
+	// the cluster IDs that will be registered in the Cluster CRDs by submariner
+	framework.TestContext.ClusterIDs = []string{"ClusterA", "ClusterB"}
+
+	config.GinkgoConfig.FocusString = "dataplane"
+	config.DefaultReporterConfig.Verbose = verboseConnectivityVerification
+	config.DefaultReporterConfig.SlowSpecThreshold = 60
+}
+
+func checkValidateArguments(args []string) error {
+	if len(args) < 2 {
+		return fmt.Errorf("validate needs exactly two kubeconfigs to validate connectivity between two clusters")
+	}
+	if connectionAttempts < 1 {
+		return fmt.Errorf("--connection-attempts must be >=1")
+	}
+
+	if connectionTimeout < 60 {
+		return fmt.Errorf("--connection-timeout must be >=60")
```

Comment:
Why >= 60?

Reply:
Because if it's too slow it wont work, since we start the "listener pod", until we go to the other cluster, and ask the "connector pod" to start... takes long (long pod startup time...) it will fail, but it won't be submariner's fault.

So we need to keep that timeout high. It's something that we could improve at some point in the framework.

Reply:
May be the listener timeout should be different and longer... but that's something to improve framework side.